sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/FluxByteBufferContent.java

```diff
@@ -89,7 +100,36 @@ public Flux<ByteBuffer> toFluxByteBuffer() {
 
     @Override
     public boolean isReplayable() {
-        return false;
+        return isReplayable;
+    }
+
+    @Override
+    public BinaryDataContent toReplayableContent() {
+        Flux<ByteBuffer> bufferedFlux = content
+            .map(buffer -> {
+                if (buffer.isDirect()) {
+                    // deep copy direct buffers, they might be pooled memory or mapped file.
+                    ByteBuffer copy = ByteBuffer.allocate(buffer.remaining());
+                    copy.put(buffer);
+                    copy.flip();
+                    return copy;
+                } else {
+                    // these are heap buffers, safe to retain.
+                    return buffer;
+                }
+            })
+            // collectList() uses ArrayList.
+            .collect(LinkedList::new, (BiConsumer<LinkedList<ByteBuffer>, ByteBuffer>) LinkedList::add)
+            .cache()
+            .flatMapMany(
+                // Duplicate buffers on re-subscription.
+                listOfBuffers -> Flux.fromIterable(listOfBuffers).map(ByteBuffer::duplicate));
+        return new FluxByteBufferContent(bufferedFlux, length, true);
```

Comment:
Should we do this once and not create new instances of FluxByteBufferContent if `toReplayableContent` is called multiple times?

Reply:
yup, that was overlook. fixed.